--- origsrc/appstream-glib-0.2.8/configure.ac	2014-11-10 10:27:11.000000000 -0600
+++ src/appstream-glib-0.2.8/configure.ac	2015-01-04 03:16:48.934937800 -0600
@@ -62,8 +62,6 @@ AM_PROG_CC_C_O
 GETTEXT_PACKAGE=appstream-glib
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Package name for gettext])
-AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.17])
 IT_PROG_INTLTOOL([0.40.0])
 
 # set up gtk-doc
